### PR TITLE
start work on player muting as described in #418

### PR DESCRIPTION
**Pull request by mithrand0**

### CODE REVIEW

File: src/game/server/player.cpp [hunk 1]
```diff
@@ -9522,3 +9522,8 @@ void CBasePlayer::UpdateFXVolume( void )
 	}
 }
 
+bool CBasePlayer::CanSpeak(void) {
```
Comment (BenLubar) [inline src/game/server/player.cpp:9525]: this should be an override in CASW_Player

File: src/game/shared/swarm/asw_gamerules.h [hunk 2]
```diff
@@ -571,6 +574,9 @@ class CAlienSwarm : public CSingleplayRules
 #ifndef CLIENT_DLL
 	IASW_Map_Builder *m_pMapBuilder;
 	STEAM_CALLBACK( CAlienSwarm, OnSteamRelayNetworkStatusChanged, SteamRelayNetworkStatus_t );
+
+	KeyValues* playersMuted;
```
Comment (BenLubar) [inline src/game/shared/swarm/asw_gamerules.h:578]: Use hungarian notation (m_pPlayersMuted) so it's clearer that this isn't a local variable

File: src/game/shared/swarm/asw_gamerules.cpp [hunk 3]
```diff
@@ -9782,3 +9801,26 @@ void CAlienSwarm::SetPingLocation( const SteamNetworkPingLocation_t & location )
...
+
+bool CAlienSwarm::IsPlayerMuted(CASW_Player* pPlayer)
+{
+	// get steamid of player
+	IPlayerInfo* pInfo = pPlayer->GetPlayerInfo();
+	int slot = pInfo->GetUserID();
+
+	CSteamID steamIDForPlayer(pInfo->GetUserID(), 1, SteamUtils()->GetConnectedUniverse(), k_EAccountTypeIndividual);
+
+	ConMsg("steamid: %d\n", steamIDForPlayer.ConvertToUint64());
```
Comment (BenLubar) [inline src/game/shared/swarm/asw_gamerules.cpp:9813]: int64 is `%llu`, not `%d`

**Comment by mithrand0** [comment]
closing this, reason: not enough focus and time right now

**Pull request closed without merging**